astroid/inference.py

```diff
@@ -120,7 +118,9 @@ def infer_sequence(
 nodes.Set._infer = infer_sequence  # type: ignore[assignment]
 
 
-def infer_map(self, context=None):
+def infer_map(
+    self: _NodesDictT, context: InferenceContext | None = None
+) -> Iterator[_NodesDictT | nodes.Dict]:
```

Comment:
Is being explicit like this desired? Or should we just return `Iterator[nodes.Dict]`?